### ISSUE TITLE
Cannot read property '_elementRef' of undefined while using custom element

### ISSUE THREAD

**Issue by osramesh006** [bfwg/ngx-drag-scroll#233]
<drag-scroll >
                  <app-course-card class="col-xl pl-2 pr-2" [card]="data" [title]="'mostViewed'" [isType]="'course'"
                    *ngFor="let data of mostViewed; let i = index" drag-scroll-item></app-course-card>
                </drag-scroll>

using version ::

**   "ngx-drag-scroll": "7.3.2",**

DiscoverPageComponent.html:12 ERROR TypeError: Cannot read property '_elementRef' of undefined
    at DragScrollComponent.push../node_modules/ngx-drag-scroll/fesm5/ngx-drag-scroll.js.DragScrollComponent.adjustMarginToLastChild (ngx-drag-scroll.js:1100)
    at DragScrollComponent.push../node_modules/ngx-drag-scroll/fesm5/ngx-drag-scroll.js.DragScrollComponent.ngAfterViewInit (ngx-drag-scroll.js:303)
    at callProviderLifecycles (core.js:22416)
    at callElementProvidersLifecycles (core.js:22390)
    at callLifecycleHooksChildrenFirst (core.js:22380)
    at checkAndUpdateView (core.js:23316)
    at callViewAction (core.js:23548)
    at execComponentViewsAction (core.js:23490)
    at checkAndUpdateView (core.js:23313)
    at callViewAction (core.js:23548)

**Comment by RobertCrowdis** [comment]
Also having this issue when assigning it to an <img> tag inside of two <ng-container> elements, 7.3.2. Going to try and find a solution..

**Comment by RobertCrowdis** [comment]
After a big of debugging, I found that this can be avoided, at least in my case. It probably has to do with the way drag-scroll-item is found in nested elements - it doesn't behave very well.

In my case I started with this:

`
  <drag-scroll #carousel>
    <ng-container *ngIf="candyList">
      <ng-container *ngFor="let candy of candyList">
        <img drag-scroll-item [src]="candy.img">
      </ng-container>
    </ng-container>
  </drag-scroll>
`

And ended up with this, throwing no errors:

`
  <drag-scroll #carousel *ngIf="candyList">
      <ng-container *ngFor="let candy of candyList">
        <img drag-scroll-item [src]="candy.img">
      </ng-container>
  </drag-scroll>
`

For the above code, I will need to add a new element to replace the hidden carousel if candyList is empty, but I don't know if I plan to do that.
